examples/color_palette/src/main.rs

```diff
@@ -0,0 +1,643 @@
+                                chroma,
+                                ..lch
+                            })
+                        },
+                    ))
+                    .push(Slider::new(
+                        lch3,
+                        0.0..=360.0,
+                        lch.hue.to_positive_degrees(),
+                        move |hue| {
+                            Message::LchColorChanged(palette::Lch {
+                                hue: palette::LabHue::from_degrees(hue),
+                                ..lch
+                            })
+                        },
+                    ))
+                    .push(
+                        TextInput::new(
+                            &mut self.lch_text_state,
+                            "",
+                            &mut self.lch_text_value,
+                            |_s| Message::TextInput,
+                        )
+                        .width(Length::Units(150))
+                        .size(14)
+                        .padding(2),
+                    ),
+            )
+            .push(
+                Canvas::new()
+                    .width(Length::Fill)
+                    // .height(Length::Units(250))
+                    .height(Length::Fill)
+                    .push(self.canvas_layer.with(&self.state)),
+            )
+            .into()
+    }
+}
+
+impl State {
+    pub fn new() -> State {
+        let base = Color::from_rgb8(75, 128, 190);
+        State {
+            color: base,
+            theme: generate_theme(&base),
+        }
+    }
+}
+
+impl canvas::Drawable for State {
+    fn draw(&self, frame: &mut canvas::Frame) {
+        use canvas::{Fill, Path};
+        use iced::{HorizontalAlignment, VerticalAlignment};
+        use palette::{Hsl, Srgb};
+
+        if self.theme.len() == 0 {
+            println!("Zero len");
+            return;
+        }
+
+        let pad = 20.0;
+
+        let box_width = frame.width() / self.theme.len() as f32;
+        let box_height = frame.height() / 2.0 - pad;
+
+        let mut text = canvas::Text::default();
+        text.horizontal_alignment = HorizontalAlignment::Center;
+        text.vertical_alignment = VerticalAlignment::Top;
+        text.size = 15.0;
+
+        for i in 0..self.theme.len() {
+            let anchor = Point {
+                x: (i as f32) * box_width,
+                y: 0.0,
+            };
+            let rect = Path::new(|path| {
+                path.move_to(anchor);
+                path.line_to(Point {
+                    x: anchor.x + box_width,
+                    y: anchor.y,
+                });
+                path.line_to(Point {
+                    x: anchor.x + box_width,
+                    y: anchor.y + box_height,
+                });
+                path.line_to(Point {
+                    x: anchor.x,
+                    y: anchor.y + box_height,
+                });
+            });
```

Comment:
I added [`Path::rectangle`](https://docs.rs/iced_wgpu/0.2.1/iced_wgpu/widget/canvas/path/struct.Path.html#method.rectangle) fairly recently. Maybe it could be used to simplify some of this code.

Reply:
Switched to use `rectangle` in 3e71eaee37bc3aea85feb0f643dcbd4ecc11d0c4